index.ts

```diff
@@ -154,6 +154,7 @@ app.get("/qr/:type/:room?", async (req: express.Request, res: express.Response)
 });
 
 app.all("/stimmung/:room?", checkAuth, function (req: express.Request, res: express.Response) {
+  req.session.participate = true;
```

Comment:
Putting this into the session makes it global to the client (browser) and by this making it vulnerable to all kind of race conditions, i.e. what if the user opens a participation link and a read-only link to another room at the same time?

So this flag should not be bound to the session - which is one per browser - but to the websocket connection - which is one per open tab/page.

Reply:
How about adding a switch to the [`wsUrl` in `cards.js`](https://github.com/BDSU/meeting-moderation-cards/blob/master/js/cards.js#L11-L15):

```diff
  let wsUrl =
    (location.protocol == "https:" ? "wss://" : "ws://") +
    location.hostname +
    (location.port ? ":" + location.port : "") +
-   '/' + roomId;
+   '/' + (participate ? 'participate' : 'watch') + '/' + roomId;